ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -776,18 +787,49 @@ def _useRealization(self, info, rlz):
 
       # 2 @ n: Crossover from set of parents
       # Create childrenCoordinates (x1,...,xM)
+      # if crossover probability is a float, keep it as is. But, If it's a string, called appropriate function.
+    if(self._crossoverProbType== "static"):
       childrenXover = self._crossoverInstance(parents=parents,
                                               variables=list(self.toBeSampled),
                                               crossoverProb=self._crossoverProb,
                                               points=self._crossoverPoints)
-
+    elif(self._crossoverProb == "linear"):
+        self._crossoverProb = adaptiveLinearCrossoverProbability(self.getIteration(traj),self.limit)
+        childrenXover = self._crossoverInstance(parents=parents,
+                                              variables=list(self.toBeSampled),
+                                              crossoverProb=self._crossoverProb,
+                                              points=self._crossoverPoints)
+        self._crossoverProb = "linear"
```

Comment:
Instead of assigning this to self._cross.. then overriding it again to be 'linear' you can do:
`crossoverProb = adaptiveLinearCrossoverProbability(self.getIteration(traj),self.limit)`
then remove 802